### PR TITLE
feat: use react-compiler to optimise render

### PR DESCRIPTION
**Pull request by stipsan**
[Test PR in the Studio](https://github.com/sanity-io/sanity/pull/7706)

### CODE REVIEW

File: packages/editor/package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@portabletext/editor",
-  "version": "1.1.11",
+  "version": "1.1.12-canary.1",
```
Comment (christianhg) [inline packages/editor/package.json:3]: Why canary? What complications will this bring to the ordinary release process?

File: packages/editor/package.json [hunk 2]
```diff
@@ -91,10 +93,16 @@
...
     "@types/react-dom": "^18.3.1",
     "@types/ws": "~8.5.12",
+    "@typescript-eslint/eslint-plugin": "^8.12.2",
+    "@typescript-eslint/parser": "^8.12.2",
     "@vitejs/plugin-react": "^4.3.3",
     "@vitest/browser": "^2.1.4",
     "@xstate/react": "^4.1.3",
+    "babel-plugin-react-compiler": "beta",
     "dotenv": "^16.4.5",
+    "eslint": "8",
```
Comment (christianhg) [inline packages/editor/package.json:103]: Guess who's back 😂

File: packages/editor/package.json (hunk 1 above)
Comment (stipsan) [reply packages/editor/package.json:3]: No complications, release please ignores the version used here and follows the one in the manifest.
I kept it in the version field for transparency on the fact that I published a canary of this PR 😌

File: packages/editor/package.json (hunk 2 above)
Comment (stipsan) [reply packages/editor/package.json:103]: Well, it doesn't seem like biome does a good job validating useEffect deps? 😅

**Comment by christianhg** [comment]
Is it perhaps a bit incorrect to use `feat` for this commit? As far as I can tell, the PR enables the compiler for the Playground and the tests. Or did I misunderstand something?:

File: packages/editor/package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "@portabletext/editor",
-  "version": "1.2.0",
+  "version": "1.1.12-canary.3",
```
Comment (christianhg) [inline packages/editor/package.json:3]: Do we want to merge with this change? I don't understand...

**Review by christianhg — CHANGES_REQUESTED** [review]
I left some doubts in the comments so marking this as "Request changes"

**Review by christianhg — APPROVED** [review]

File: packages/editor/package.json (hunk 3 above)
Comment (stipsan) [reply packages/editor/package.json:3]: Release please will correct it later so it's fine :)

**Comment by christianhg** [comment]
LFG

**Pull request merged**